third_party/amd/lib/TritonAMDGPUTransforms/OptimizeEpilogue.cpp

```diff
@@ -38,11 +38,38 @@ using namespace mlir;
 
 namespace {
 
+bool isOneOperandElementwiseOp(Operation *op) {
+  if (llvm::isa<arith::ExtFOp, arith::ExtSIOp, arith::ExtUIOp, arith::FPToSIOp,
+                arith::FPToUIOp, arith::NegFOp, arith::SIToFPOp,
+                arith::TruncFOp, arith::TruncIOp, arith::UIToFPOp>(op))
+    return true;
+  if (llvm::isa<math::AbsFOp, math::AbsIOp, math::AtanOp, math::Atan2Op,
+                math::CeilOp, math::CosOp, math::SinOp,
+                math::CountLeadingZerosOp, math::CountTrailingZerosOp,
+                math::CtPopOp, math::ErfOp, math::ExpOp, math::Exp2Op,
+                math::ExpM1Op, math::FloorOp, math::LogOp, math::Log10Op,
+                math::Log1pOp, math::Log2Op, math::RsqrtOp, math::SqrtOp,
+                math::TanhOp>(op))
+    return true;
+  if (llvm::isa<triton::IntToPtrOp, triton::PtrToIntOp, triton::BitcastOp,
+                triton::FpToFpOp>(op))
+    return true;
+  if (auto externElementwiseOp = dyn_cast<triton::ExternElementwiseOp>(op))
+    return externElementwiseOp.getPure();
```

Comment:
this may have more than 1 operand.
Would check the traits Elementwise + numOperands() == 1 + isPure(op) handle all the cases?